sqlserver/tests/test_activity.py

```diff
@@ -0,0 +1,252 @@
+    instance_docker['query_metrics'] = {'enabled': False}
+    return copy(instance_docker)
+
+
+@pytest.fixture
+def instance_sql_msoledb_dbm(instance_sql_msoledb):
+    instance_sql_msoledb['dbm'] = True
+    instance_sql_msoledb['min_collection_interval'] = 1
+    instance_sql_msoledb['query_activity'] = {'enabled': True, 'run_sync': True, 'collection_interval': 0.1}
+    # not needed for this test
+    instance_sql_msoledb['query_metrics'] = {'enabled': False}
+    instance_sql_msoledb['tags'] = ['optional:tag1']
+    return instance_sql_msoledb
+
+
+@not_windows_ci
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+def test_collect_activity(aggregator, instance_docker, dd_run_check, dbm_instance):
+    _run_test_collect_activity(aggregator, instance_docker, dd_run_check, dbm_instance)
+
+
+@windows_ci
+@pytest.mark.integration
+def test_collect_activity_windows(aggregator, instance_docker, dd_run_check, instance_sql_msoledb_dbm):
+    _run_test_collect_activity(aggregator, instance_docker, dd_run_check, instance_sql_msoledb_dbm)
+
+
+def _run_test_collect_activity(aggregator, instance_docker, dd_run_check, dbm_instance):
+    check = SQLServer(CHECK_NAME, {}, [dbm_instance])
+    query = "select * from things"
+    bob_conn = _get_conn_for_user(instance_docker, "bob")
+    with bob_conn.cursor() as cursor:
+        cursor.execute("USE {}".format("datadog_test"))
+        cursor.execute(query)
+
+    fred_conn = _get_conn_for_user(instance_docker, "fred")
+    with fred_conn.cursor() as cursor:
+        cursor.execute("USE {}".format("datadog_test"))
+        cursor.execute(query)
+
+    time.sleep(2)  # wait for coll interval
+    dd_run_check(check)
+    fred_conn.close()  # close the open tx that belongs to fred
+    time.sleep(2)  # wait for coll interval
+    dd_run_check(check)  # run check again
+    expected_instance_tags = set(dbm_instance.get('tags', []))
+
+    dbm_activity = aggregator.get_event_platform_events("dbm-activity")
+    assert len(dbm_activity) == 2, "should have collected exactly two dbm-activity payloads"
+    first = dbm_activity[0]
+    # validate common host fields
+    assert first['host'] == "stubbed.hostname", "wrong hostname"
+    assert first['dbm_type'] == "activity", "wrong dbm_type"
+    assert first['ddsource'] == "sqlserver", "wrong source"
+    assert first['ddagentversion'], "missing ddagentversion"
+    assert set(first['ddtags']) == expected_instance_tags, "wrong instance tags activity"
+    assert type(first['collection_interval']) in (float, int), "invalid collection_interval"
+
+    assert len(first['sqlserver_activity']) > 0
+    # bob and fred's open transactions should have been
+    # collected on first iteration. and bob's has been open longer
+    # so it should come first in the payload
+    first_users = [f['user_name'] for f in first['sqlserver_activity']]
+    bobs_row = first['sqlserver_activity'][0]
+    assert "bob" and "fred" in first_users
```

Comment:
This isn't actually checking whether bob is in there 

```
>>> "bob" and "fred" in {"fred"}
True
```

Reply:
ugh python